### PR TITLE
Issue #5093 Static UrlEncoded

### LINKED ISSUE #5093: Review UrlEncoded locking
Spawned by #5083.

**Jetty version**
9.4.x

**Description**
`UrlEncoded` needs a serious review.

It is used almost only for its `static` methods, however is itself a `MultiMap` and can decode a query string into itself, which is only used in tests.

The `static` methods return void but require a `MultiMap` parameter that is always created anew, but it is used as a lock inside the `UrlEncoded` methods, which is completely unnecessary.

The whole locking of this class should be removed, and the `static` methods of this class should return a `MultiMap` or a `Map`.

Also, it should just be a static utility class, rather than inheriting from `MultiMap`.

### PR DESCRIPTION
**Pull request by gregw**
Updated UrlEncoded to static only class with no synchronization for #5093

### CODE REVIEW

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java [hunk 1]
```diff
@@ -835,13 +836,13 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
             assertEquals(null, request.getPathTranslated());
 
             UrlEncoded query = new UrlEncoded();
```
Comment (joakime) [inline jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java]: Shouldn't this now be?
``` java
MultiMap<String> query = new MultiMap<>();
```

How did this compile with `new UrlEncoded()` now being private?

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java [hunk 2]
```diff
@@ -835,13 +836,13 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
-            query.decode(request.getQueryString());
+            UrlEncoded.decodeTo(request.getQueryString(), new MultiMap<>(), UrlEncoded.ENCODING);
             assertThat(query.getString("do"), is("end"));
 
             // Russian for "selected=Temperature"
             UrlEncoded q2 = new UrlEncoded();
```
Comment (joakime) [inline jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java]: Ditto..
``` java
MultiMap<String> q2 = new MultiMap<>();
```

**Review by joakime — CHANGES_REQUESTED** [review]

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java (hunk 2 above)
Comment (gregw) [reply jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java]: yep! oops!

**Review by joakime — APPROVED** [review]
LGTM

**Pull request merged**